CLI Typegen doesn't work with imported partial queries

**Describe the bug**

When using groq typegen `sanity schema extract && sanity typegen generate` it is perfectly able to work with queries that are composed out of concatenated string variables. It, however doesn't work, crashes, if any of the variables are imported from another file.

**To Reproduce**

1. Create 2 files - `repo.ts` and `util.ts`
2. Place the following query into the `repo.ts`:

   ```groq
   import { inner } from "./util";
   
   export function marko() {
       const myTestQuery = groq`*[_type == "sanity.imageAsset"] ${inner}`;
       return client.fetch(myTestQuery);
   }
   ```
3. Place the following query segment into the `util.ts` file:

   ```groq
   export const inner = `{ metadata }`;
   ```
4. Run the typegen command: `sanity schema extract && sanity typegen generate`
5. It will crash with an error, saying that it `✖ Cannot find module './util'`

**Expected behavior**

It should behave the same as it does when the inner variable is within the same file as the query - successfully compose the query and its type.

**Which versions of Sanity are you using?**

```none
@sanity/cli (global) 3.37.1 (up to date)
@sanity/document-internationalization 3.0.0 (up to date)
@sanity/icons 2.11.8 (up to date)
@sanity/image-url 1.0.2 (up to date)
@sanity/presentation 1.12.3 (up to date)
@sanity/ui 2.1.1 (up to date)
@sanity/vision 3.37.1 (up to date)
@sanity/webhook 4.0.3 (up to date)
sanity 3.37.1 (up to date)
```

**What operating system are you using?**

macOS Sonoma

**Which versions of Node.js / npm are you running?**

```none
10.2.4
v20.11.1
```

--- a/packages/@sanity/codegen/src/typescript/__tests__/findQueriesInSource.test.ts
+++ b/packages/@sanity/codegen/src/typescript/__tests__/findQueriesInSource.test.ts
@@ -1,5 +1,3 @@
-import path from 'node:path'
-
 import {describe, expect, test} from '@jest/globals'
 
 import {findQueriesInSource} from '../findQueriesInSource'
@@ -81,18 +79,19 @@ describe('findQueries', () => {
       const postQuery = groq\`*[_type == "\${foo}"]\`
       const res = sanity.fetch(postQueryResult);
     `
+    const queries = findQueriesInSource(source, __filename, undefined)
+    expect(queries.length).toBe(1)
+    expect(queries[0].result).toBe('*[_type == "foo"]')
+  })
 
-    const resolver: NodeJS.RequireResolve = (id) => {
-      if (id === 'foo') {
-        return path.resolve(__dirname, 'fixtures', 'exportVar')
-      }
-      return require.resolve(id)
-    }
-    resolver.paths = (request: string): string[] | null => {
-      return require.resolve.paths(request)
-    }
-
-    const queries = findQueriesInSource(source, 'test.ts', undefined, resolver)
+  test('should import, subdirectory', () => {
+    const source = `
+      import { groq } from "groq";
+      import {foo}  from "../__tests__/fixtures/exportVar";
+      const postQuery = groq\`*[_type == "\${foo}"]\`
+      const res = sanity.fetch(postQueryResult);
+    `
+    const queries = findQueriesInSource(source, __filename, undefined)
     expect(queries.length).toBe(1)
     expect(queries[0].result).toBe('*[_type == "foo"]')
   })

--- a/packages/@sanity/codegen/src/typescript/expressionResolvers.ts
+++ b/packages/@sanity/codegen/src/typescript/expressionResolvers.ts
@@ -323,12 +323,14 @@ function resolveImportSpecifier({
     throw new Error(`Could not find import declaration for ${node.local.name}`)
   }
 
-  const importName = node.local.name
-  const importFileName = importDeclaration.source.value
-  // const importPath = path.resolve(path.dirname(filename), importFileName);
-  const importPath = importName.startsWith('./')
-    ? path.resolve(path.dirname(filename), importFileName)
-    : importFileName
+  const importName = node.local.name // the name of the variable to import
+  const importFileName = importDeclaration.source.value // the file to import from
+
+  const importPath =
+    importFileName.startsWith('./') || importFileName.startsWith('../')
+      ? path.resolve(path.dirname(filename), importFileName)
+      : importFileName
+
   const resolvedFile = resolver(importPath)
   const source = fs.readFileSync(resolvedFile)
   const tree = parseSourceFile(source.toString(), resolvedFile, babelConfig)

